app/src/main/java/com/kamwithk/ankiconnectandroid/ankidroid_api/NoteAPI.java

```diff
@@ -73,9 +77,25 @@ public boolean updateNoteFields(long note_id, String[] fields) throws Exception
         return api.updateNoteFields(note_id, fields);
     }
 
-    public long getNoteModelId(long note_id) throws Exception {
-        api.getNote(note_id); //set cursor to correct position
-        return api.getCurrentModelId();
+    public Long getNoteModelId(long note_id) {
+        // Manually queries the note with a specific projection to get the model ID
```

Comment:
Why manually copy paste the code here instead of using their API?
Was there something wrong with it? Ideally we use their API in case anything  on their end changes (but at the same time it isn't too much code)

Reply:
After comparing the new vs old code and looking at what the Anki Android functions do, I now realise you are directly getting the note model id here instead of getting it from the note